environment.yml

```diff
@@ -89,6 +89,7 @@ dependencies:
   - flake8-bugbear=22.7.1 # used by flake8, find likely bugs
   - isort>=5.2.1  # check that imports are in the right order
   - mypy=0.981
+  - pylint=2.15.5
```

Comment:
@MarcoGorelli is there a reason to add pylint to the env file? The check is still run through pre-commit, which will make install it in its own env?

(although that's also the case for black et al, and those are also here, so it's all OK, just wondering)

Reply:
`pylint` can do some extra inference if it has everything built (hence a few extra checks that need to be turned off in this PR) - if it's only running in CI, then we might as well use the environment where pandas has been built (like we do for `mypy`, that one also has `language: system`)

Reply:
Ah, I missed the `language: system` . Thanks for the explanation!

Reply:
🤔 though `language: system` is less reliable, so maybe this should still run in its own little env...yeah let's do that, then the risk of it annoying contributors / maintainers is reduced